core/vm/evm_zkevm.go

```diff
@@ -104,7 +124,7 @@ func (evm *EVM) createZkEvm(caller ContractRef, codeAndHash *codeAndHash, gas ui
 	contract.SetCodeOptionalHash(&address, codeAndHash)
 
 	// zkevm: ensure we set the IsCreate flag here so that zk virtual counters get the correct context
-	contract.IsCreate = true
+	// contract.IsCreate = true
```

Comment:
you are spot on here, after all the discussions around isDeploy etc. etc. I had the context all wrong for isCreate.  Please could you remove the lines rather than comment them out to save future confusion.